clang/runtime/dpct-rt/include/atomic_ref_ext.hpp.inc

```diff
@@ -0,0 +1,299 @@
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+//==---- atomic_ref_ext.hpp -------------------------------*- C++ -*----------------==//
+//
+// Copyright (C) Intel Corporation
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+// DPCT_LABEL_END
+
+#ifndef __DPCT_ATOMIC_REF_EXT_HPP__
+#define __DPCT_ATOMIC_REF_EXT_HPP__
+
+// DPCT_COMMENT
+// DPCT_COMMENT Example1:
+// DPCT_COMMENT // DPCT_LABEL_BEGIN|FeatureNameDef|[Namespace]
+// DPCT_COMMENT // DPCT_DEPENDENCY_EMPTY
+// DPCT_COMMENT // DPCT_CODE
+// DPCT_COMMENT some code
+// DPCT_COMMENT // DPCT_LABEL_END
+// DPCT_COMMENT
+// DPCT_COMMENT Example2:
+// DPCT_COMMENT // DPCT_LABEL_BEGIN|FeatureNameDef|[Namespace]
+// DPCT_COMMENT // DPCT_DEPENDENCY_BEGIN
+// DPCT_COMMENT // FileID|FeatureNameRef
+// DPCT_COMMENT [// FileID|FeatureNameRef]
+// DPCT_COMMENT ...
+// DPCT_COMMENT // DPCT_DEPENDENCY_END
+// DPCT_COMMENT // DPCT_CODE
+// DPCT_COMMENT some code
+// DPCT_COMMENT // DPCT_LABEL_END
+// DPCT_COMMENT
+// DPCT_COMMENT For header file including dependency, please use predefined feature name:
+// DPCT_COMMENT   local_include_dependency: dpct helper files
+// DPCT_COMMENT   non_local_include_dependency: other header files
+
+// DPCT_LABEL_BEGIN|non_local_include_dependency|
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+#include <CL/sycl.hpp>
+#include <cassert>
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|local_include_dependency|
+// DPCT_DEPENDENCY_BEGIN
+// AtomicRefExt|atomic_ref_ext
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+// DPCT_LABEL_END
+
+namespace dpct {
+
+// DPCT_LABEL_BEGIN|atomic_ref_ext|dpct
+// DPCT_DEPENDENCY_BEGIN
+// AtomicRefExt|atomic_ref_ext
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+/// Atomic pointer/reference extension to implement standard APIs in std::atomic
+template <typename T,
+          cl::sycl::memory_order DefaultOrder = cl::sycl::memory_order::seq_cst,
+          cl::sycl::memory_scope DefaultScope = cl::sycl::memory_scope::system,
+          cl::sycl::access ::address_space Space =
+              cl::sycl::access::address_space::generic_space>
+class atomic_ref_ext {
+  T __d;
+
+public:
+  /// default memory synchronization order
+  static constexpr cl::sycl::memory_order default_read_order =
+      cl::sycl::detail::memory_order_traits<DefaultOrder>::read_order;
+  static constexpr cl::sycl::memory_order default_write_order =
+      cl::sycl::detail::memory_order_traits<DefaultOrder>::write_order;
+  static constexpr cl::sycl::memory_scope default_scope = DefaultScope;
+  static constexpr cl::sycl::memory_order default_read_modify_write_order =
+      DefaultOrder;
+// DPCT_LABEL_END
+
+// DPCT_LABEL_BEGIN|atomic_ref_ext_1|dpct
+// DPCT_DEPENDENCY_BEGIN
+// AtomicRefExt|atomic_ref_ext
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+  /// no-argument default constructor
+  atomic_ref_ext() = default;
+  /// default constructor
```

Comment:
Default constructor is constructor with no args.

Reply:
fix it in commits 34e891d3e6db93f6f638490d0805d813e0bf7879